src/jit/codegenarm.cpp

```diff
@@ -1455,7 +1533,39 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
 //
 void CodeGen::genFloatToFloatCast(GenTreePtr treeNode)
 {
-    NYI("Cast");
+    // float <--> double conversions are always non-overflow ones
+    assert(treeNode->OperGet() == GT_CAST);
+    assert(!treeNode->gtOverflow());
+
+    regNumber targetReg = treeNode->gtRegNum;
+    assert(genIsValidFloatReg(targetReg));
+
+    GenTreePtr op1 = treeNode->gtOp.gtOp1;
+    assert(!op1->isContained());               // Cannot be contained
```

Comment:
With recent changes, this should probably be `isUsedFromReg()` to be more precise. But could be done in a future change.